### PR TITLE
Fix: Lazy Loading images

### LINKED ISSUE #22542: Reinstate Jetpack / Boost Lazy-Loading sync
In #22163, we massively refactored Critical CSS's classes, and introduced a much cleaner object model. Along the way, one feature was broken - syncing Jetpack Lazy Loading with Boost Lazy Loading.

This ticket is a reminder that we need to fix this before the next release.

Also, related tests were deactivated. See #22163 for details.

### PR DESCRIPTION
**Pull request by pyronaur**
Fixes #22542

Sync Jetpack and Jetpack Boost lazy image status so that toggles in both dashboards always reflect the current status.

Modified the logic slightly. Because Jetpack is watching for filters and reflecting that to the dashboard, I decided to just use Jetpack plugin as the single source of truth instead of Jetpack Boost for the lazy images setting.

**Does this pull request change what data or activity we track or use?**
No

**Testing instructions:**
Run E2E Tests

### CODE REVIEW

File: projects/plugins/boost/compatibility/jetpack.php [hunk 1]
```diff
@@ -35,3 +35,33 @@ function exclude_jetpack_likes_scripts_defer( $exclusions ) {
...
+}
+
+add_filter( 'default_option_jetpack_boost_status_lazy-images', __NAMESPACE__ . '\lazy_images_override_status' );
+add_filter( 'option_jetpack_boost_status_lazy-images', __NAMESPACE__ . '\lazy_images_override_status' );
+
+/**
+ * Forward all lazy image settings changes to Jetpack
+ * when interacting with Jetpack Boost dashboard.
+ */
+function lazy_imasges_sync_status( $_unused, $new_value ) {
```
Comment (thingalon) [inline projects/plugins/boost/compatibility/jetpack.php:56]: `imasges`

**Review by thingalon — CHANGES_REQUESTED** [review]
Minor thing to fix, but otherwise looks good. Hadn't thought of using a compatibility module. That's not a bad idea.

Don't forget to use the full Jetpack Monorepo PR template, to assure the gardening and labelling system you've thought about its questions.

File: projects/plugins/boost/compatibility/jetpack.php (hunk 1 above)
Comment (pyronaur) [reply projects/plugins/boost/compatibility/jetpack.php]: derp 🤪

**Review by thingalon — APPROVED** [review]
Let's do it.

**Pull request merged**